### ISSUE TITLE
Missing restore from index on C bindings

### ISSUE THREAD

**Issue by theolivenbaum** [facebook/rocksdb#7087]
The [How to backup RocksDB](https://github.com/facebook/rocksdb/wiki/How-to-backup-RocksDB%3F) wiki page mentions the following function that can be used to restore from a specific backup index:

`````cpp
backup_engine->RestoreDBFromBackup(1, "/tmp/rocksdb", "/tmp/rocksdb");
`````

But this seems to be missing from the C ABI, which exposes only [the method to restore from the latest index](https://github.com/facebook/rocksdb/blob/1b85d57cf5a2b41d6a026557adfb245d5235d4aa/include/rocksdb/c.h#L170)

`````cpp
void rocksdb_backup_engine_restore_db_from_latest_backup(
    rocksdb_backup_engine_t* be, const char* db_dir, const char* wal_dir,
    const rocksdb_restore_options_t* restore_options, char** errptr) {
  SaveError(errptr, be->rep->RestoreDBFromLatestBackup(std::string(db_dir),
                                                       std::string(wal_dir),
                                                       restore_options->rep));
}
`````

Is there any reason this is not exposed on the ABI?

**Comment by ajkr** [comment]
Agreed it is missing. Would you be interested in contributing it?

**Comment by theolivenbaum** [comment]
Sure, I'll send a PR for it then!

**Comment by theolivenbaum** [comment]
@ajkr: here you have it: https://github.com/facebook/rocksdb/pull/7092 
Took the opportunity to add also a link to [our fork of the C# package](https://github.com/curiosity-ai/rocksdb-sharp) that is more up-to-date than the one mentioned previously.

**Comment by theolivenbaum** [comment]
It seems like the PR now passes all CircleCI tests after fixing some minor formatting issues 😄 

@ajkr any chance to get this merged sometime soon? this is a blocker for us to finish with our internal backup/restore work...

**Comment by ajkr** [comment]
It is merged now.

**Comment by theolivenbaum** [comment]
Hi @ajkr - just noticed this one was missing from yesterday's [v6.11.4](https://github.com/facebook/rocksdb/releases/tag/v6.11.4) release , any chance it can get scheduled for the next one? Thanks!

**Comment by ajkr** [comment]
Right, this PR landed after the 6.11 branch got cut (~3 weeks ago) and since then we only backported bug fixes. This will definitely be in the 6.12 branch though as that branch is yet to be cut.
